internal/pipeline/pipeline_invariants.go

```diff
@@ -5,40 +5,35 @@ import (
 	"strings"
 )
 
-type PipelineInvariants struct {
-	OrganizationSlug string
-	PipelineSlug     string
-	Repository       string
-}
-
-var _ SignedFielder = (*CommandStepWithPipelineInvariants)(nil)
+var _ SignedFielder = (*CommandStepWithInvariants)(nil)
 
-type CommandStepWithPipelineInvariants struct {
+// CommandStepWithInvariants is a CommandStep with PipelineInvariants.
+type CommandStepWithInvariants struct {
 	CommandStep
-	PipelineInvariants
+	RepositoryURL string
 }
 
 // SignedFields returns the default fields for signing.
-func (c *CommandStepWithPipelineInvariants) SignedFields() (map[string]any, error) {
+func (c *CommandStepWithInvariants) SignedFields() (map[string]any, error) {
 	return map[string]any{
-		"command":             c.Command,
-		"env":                 c.Env,
-		"plugins":             c.Plugins,
-		"matrix":              c.Matrix,
-		"pipeline_invariants": c.PipelineInvariants,
+		"command":        c.Command,
+		"env":            c.Env,
+		"plugins":        c.Plugins,
+		"matrix":         c.Matrix,
+		"repository_url": c.RepositoryURL,
```

Comment:
FWIW, putting each invariant into its own signed field is closer to what I had in mind with signed-fields originally (if/when we add more of them, we'll be backwards compatible).

Reply:
Yeah, I agree. One "pipeline_invariants" field was a mistake. I like the ergonomics of passing them all around in a single struct, but for the signed fields, they should be flattened out.

You may have seen me talk about a way to "compose signedFielders" in the chats somewhere. The idea that you could pass the invariants around in a struct, but the signedFielder implementation would flatten and prefix them appropriately. It's not necessary to do this right now, though, so I didn't.